docs: add gosling api to readme (#8)
gosling-lang/gosling-react: README.md

rpm run start
```

## Gosling API

> To use the Gosling API, we need to create a reference to the GoslingComponent.

Reviewer: Just added a link that might be helpful for people who are not familiar with React

```suggestion
To use the Gosling API, you need to create a [Ref](https://reactjs.org/docs/refs-and-the-dom.html) that stores a reference to the GoslingComponent.
```

```javascript
const gosRef = React.useRef(null)

<GoslingComponent
  ref = {gosRef}
  spec = {/**your gosling spec**/}
/>

if (gosRef.current) {
  // then you can use any Gosling API you want
  gosRef.current.api.exportPdf();
}
//
```

Below is an example
```javascript
import React, { useRef, useEffect } from "react";
import { GoslingComponent} from 'gosling.js';

function app(){
  const gosRef = useRef(null)
  
  useEffect(() => {
    if (gosRef.current) {
      gosRef.current.api.subscribe('click', (type, eventData) => {

>             console.warn(type, eventData.data);

Reviewer: Maybe add an comment?
```suggestion
            // print raw data that corresponds to the clicked mark
            console.warn(type, eventData.data);
```

      })
    }
    return ()=>{
      // remember to unsubscribe events
      gosRef.current?.api.unsubscribe('click');
    }
  }, [gosRef.current])

  return <div>
    <GoslingComponent
      ref = {gosRef}
      spec = {/**your gosling spec**/}
    />

>     <button type="button" onClick={()=>gosRef.current?.api.exportPdf();}>

Reviewer: ```suggestion
    <button type="button" onClick={()=>gosRef.current?.api.exportPdf()}>
```

      Export PDF
     </button>
  </div>
}
```

> Please refer to [gosling doc](http://gosling-lang.org/docs/js-api) for a complete list of Gosling API.

Reviewer: ```suggestion
Please refer to [Gosling documentation](http://gosling-lang.org/docs/js-api) for a complete list of Gosling API.
```
